# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1531322129 14400
#      Wed Jul 11 11:15:29 2018 -0400
# Branch stable
# Node ID 9f422e49be152a88e8a7ac9f8258a30a5b30b037
# Parent  07c2c42f457e07ab3b000f7aef48f63fb09e27bd
transform libiconv.dll.a to -liconv for mkoctfile (bug #53920)

* configure.ac (MKOCTFILE_GNULIB_DEPS): Use $MKOCTFILE_LIBICONV
instead of $LIBICONV.
(MKOCTFILE_LIBICONV): New temporary variable.  On mingw systems,
define by transforming libiconv.dll.a to -liconv in the LIBICONV
variable defined by gnulib.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -2807,17 +2807,33 @@ if test $octave_cv_gcc_has_pragma_GCC_di
     [AC_DEFINE(HAVE_WARN_IMPLICIT_FALLTHROUGH, 1,
       [Define to 1 if GCC supports -Wimplicit-fallthrough warning])])
 fi
 
 ### These are recommended by the gnulib bootstrap script:
 
 GNULIB_LINK_DEPS="$FREXPF_LIBM $FREXP_LIBM $GETHOSTNAME_LIB $LIBSOCKET $LIB_CLOCK_GETTIME $LIB_CRYPTO $LIB_GETLOGIN $LIB_NANOSLEEP $LIB_SELECT $LTLIBICONV $LTLIBINTL $LTLIBTHREAD"
 
-MKOCTFILE_GNULIB_LINK_DEPS="$FREXPF_LIBM $FREXP_LIBM $GETHOSTNAME_LIB $LIBSOCKET $LIB_CLOCK_GETTIME $LIB_CRYPTO $LIB_GETLOGIN $LIB_NANOSLEEP $LIB_SELECT $LIBICONV $LIBINTL $LIBTHREAD"
+### FIXME: This is a kluge to transform $libdir/libiconv.dll.a to -liconv.
+### It would probably be better to fix gnulib to not give us an absolute
+### file name...
+
+case $host_os in
+  mingw*)
+    ## Don't use $libdir here because that's defined to be
+    ## ${exec_prefix}/lib and at this point, ${exec_prefix} is "NONE"!
+    tmp_libiconv_dll="$LIBICONV_PREFIX/lib/libiconv.dll.a"
+    MKOCTFILE_LIBICONV=`echo $LIBICONV | sed "s|$tmp_libiconv_dll|-liconv|"`
+  ;;
+  *)
+    MKOCTFILE_LIBICONV="$LIBICONV"
+  ;;
+esac
+
+MKOCTFILE_GNULIB_LINK_DEPS="$FREXPF_LIBM $FREXP_LIBM $GETHOSTNAME_LIB $LIBSOCKET $LIB_CLOCK_GETTIME $LIB_CRYPTO $LIB_GETLOGIN $LIB_NANOSLEEP $LIB_SELECT $MKOCTFILE_LIBICONV $LIBINTL $LIBTHREAD"
 
 AC_SUBST(GNULIB_LINK_DEPS)
 
 BASE_LIBOCTAVE_LINK_DEPS="$CURL_LIBS $SPARSE_XLIBS $ARPACK_LIBS $QRUPDATE_LIBS $FFTW_XLIBS $LAPACK_LIBS $BLAS_LIBS $READLINE_LIBS $TERM_LIBS $LIBGLOB $PCRE_LIBS $DL_LIBS $PTHREAD_LIBS $FLIBS $LIBS"
 
 LIBOCTAVE_LINK_DEPS="$BASE_LIBOCTAVE_LINK_DEPS $GNULIB_LINK_DEPS"
 
 MKOCTFILE_LIBOCTAVE_LINK_DEPS="$BASE_LIBOCTAVE_LINK_DEPS $MKOCTFILE_GNULIB_LINK_DEPS"
